File: chapter_2/exercise_3/quote/lib/quote.ex
defmodule Quote do
  @moduledoc """
  Documentation for Quote.
  """

  def quote do
    get_input()
    |> output_result
  end

  defp get_input() do
    {get_quote(), get_who()}
  end

  defp output_result({qte, who}) do
    who <> " says, \"" <> qte <> "\""
    |> IO.puts
  end

  defp get_quote() do
    do_get_input("What is the quote? ")
  end

  defp get_who() do
    do_get_input("Who said it? ")
  end

  defp do_get_input(prompt) do
    IO.gets(prompt)
    |> String.trim
  end
end

File: chapter_2/exercise_3/quote.c
#include <stdio.h>
#include <string.h>

int main()
{
  const int max_input = 256;
  const char newline_chars[] = "\r\n";

  char quote[max_input];
  char who[max_input];

  printf("What is the quote? ");
  fgets(quote, max_input - 1, stdin);
  quote[strcspn(quote, newline_chars)] = 0;

  printf("Who said it? ");
  fgets(who, max_input - 1, stdin);
  who[strcspn(who, newline_chars)] = 0;

  printf("%s says, \"%s\"\n", who, quote);
}
